htdocs/fourn/facture/card-rec.php

```diff
@@ -428,8 +428,13 @@
 			$tva_tx = (GETPOST('tva_tx', 'alpha') ? GETPOST('tva_tx', 'alpha') : 0);
 			$ref_fournisseur = (GETPOSTISSET('fourn_ref') ? GETPOST('fourn_ref', 'restricthtml') : '');
 		} else {
-			$idprod = GETPOST('idprod', 'int');
+			$idprod = GETPOST('idprodfournprice', 'int');
```

Comment:
Note that idprodfournprice may be
- an int when it is a supplier price id
- a string 'idprod_xxx' where xxx is the id of product (when there is no price defined on the product).

So if you make the GETPOST with int param, for the case idprod_xxx, you will get zero. 
Few lines later there is a regex that is used so we got into idprod an int for both cases.
Currently the GETPOST is not used as it is overwritten later but with your change it is used bybthe new if so you should take into account the case value is idprod_xxx